src/lib/shadow-shim-helper-rs/src/shim_event.rs

```diff
@@ -52,6 +52,11 @@ pub struct ShimEventStartReq {
     pub thread_shmem_block_to_init: ForeignPtr<ShMemBlockSerialized>,
     /// Shim pointer to be initd by Shadow. Optional.
     pub process_shmem_block_to_init: ForeignPtr<ShMemBlockSerialized>,
+
+    /// Shim pointer to be initd by Shadow. Optional.
+    /// If set, shadow will write a null-terminated path.
+    pub initial_working_dir_to_init: ForeignPtr<u8>,
+    pub initial_working_dir_to_init_len: usize,
```

Comment:
Why not put the working dir in the process shared memory rather than writing to the plugin's memory directly?

Reply:
We discussed this, and since the working dir needs to be ~4 KiB, it might not be worth putting in shared memory. So I think either way is fine, but should have some comments here explaining why it's done this way.

Reply:
Done